command/agent/approle_end_to_end_test.go

```diff
@@ -400,6 +401,53 @@ func testAppRoleEndToEnd(t *testing.T, removeSecretIDFile bool, bindSecretID boo
 	}
 }
 
+// Prior to VAULT-8518 being fixed, you were unable to delete an approle value longer than 1024 bytes
+// due to a restriction put into place by PR #14746, to prevent unbounded HMAC creation.
+// We now restrict the creation of an approle name to being maximum 4096 bytes.
+// This test validates this logic.
+func TestAppRoleLongRoleName(t *testing.T) {
+	approleName := strings.Repeat("a", 5000)
+
+	coreConfig := &vault.CoreConfig{
+		DisableMlock: true,
+		DisableCache: true,
+		Logger:       log.NewNullLogger(),
+		CredentialBackends: map[string]logical.Factory{
+			"approle": credAppRole.Factory,
+		},
+	}
+
+	cluster := vault.NewTestCluster(t, coreConfig, &vault.TestClusterOptions{
+		HandlerFunc: vaulthttp.Handler,
+	})
+
+	cluster.Start()
+	defer cluster.Cleanup()
+
+	cores := cluster.Cores
+
+	vault.TestWaitActive(t, cores[0].Core)
+
+	client := cores[0].Client
+
+	err := client.Sys().EnableAuthWithOptions("approle", &api.EnableAuthOptions{
```

Comment:
What do you think about testing the command directly with `cmd.Run`? Using the API directly might skip over some code that would be good to test here...

Reply:
What code are you worried that we'll miss? I'm not quite sure I immediately understand the impetus of this request - could you explain a bit more what might be missing?

Reply:
Oh nothing specifically (at least off the top of my head), I just noticed that the test is an end-to-end test for the command and we're using the API rather than the command directly. My assumption (without diving super deep) was that we'd miss out on some of the flow of the `cmd.Run` path, but I may be mistaken.

I'm just drawing from lessons learned with the `login` command and wanted to be sure we're testing the correct thing.

Reply:
I see - I think I have a preference to use the client construct unless we're specifically testing the command line - I think as long as it's going through the API, it's no less 'end-to-end', and it's what the other tests in this file seem to do.

My code change didn't affect the command line client in any way, so I think there's no reason to add additional testing for the command line client

Reply:
(I might be completely wrong about your request here so do say so if I was!)

Reply:
Not at all! This makes sense, just wanted to make sure!